src/js/view/view.js

```diff
@@ -320,13 +320,21 @@ function View(_api, _model) {
                             message: 'Controls failed to load',
                             reason: reason
                         });
+                    })
+                    .then(function () {
+                        _this.trigger(CONTROLS, {
+                            controls: enable
+                        });
                     });
-            } else {
-                addControls();
+                return;
             }
+            addControls();
         } else {
             _this.removeControls();
         }
+        _this.trigger(CONTROLS, {
+            controls: enable
+        });
     }
 
```

Comment:
I'm fine with triggering controls from the view, but instead of changing this method, I would prefer if you triggered the event at the end of the method. It should always be triggered regardless of whether controls are ready. You can add the promise to the event if we're dealing with async behavior.

```js
    function changeControls(model, enable) {
        const controlsEvent = {
            controls: enable
       };
        if (enable) {
            ControlsModule = ControlsLoader.module.controls;
            if (!ControlsModule) {
                controlsEvent.loadPromise = ControlsLoader.load()
                    .then(function (Controls) {
                        ControlsModule = Controls;
                        // Check that controls is still true after the loader promise resolves
                        const enabledState = model.get('controls');
                        if (enabledState) {
                            addControls();
                        }
                        return enabledState;
                    })
                    .catch(function (reason) {
                        _this.trigger(ERROR, {
                            message: 'Controls failed to load',
                            reason: reason
                        });
                        return reason;
                    });
            } else {
                addControls();
            }
        } else {
            _this.removeControls();
        }
        _this.trigger(CONTROLS, controlsEvent);
    }
```